tests/test_defectsCombined.py

```diff
@@ -0,0 +1,766 @@
+                     Box2I(corner=Point2I(50, 116), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 118), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 120), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 122), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 124), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 126), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 128), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 130), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 132), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 134), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 136), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 138), dimensions=Extent2I(1, 1)),
+                     Box2I(corner=Point2I(50, 140), dimensions=Extent2I(1, 1)),
+                     # This last point is out of order in y.
+                     Box2I(corner=Point2I(50, 100), dimensions=Extent2I(1, 1))]
+
+        # Force defect normalization off in order to trigger DM-38301, because
+        # defects.fromFootprintList() which is called by _findHotAndColdPixels
+        # does not do normalization.
+        defects._bulk_update = True
+        for badBox in badPixels:
+            defects.append(badBox)
+        defects._bulk_update = False
+
+        self.check_maskBlocks(defects, expectedDefects)
+
+    def test_defectFindingAllSensor(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderLeftRight = 0
+        config.nPixBorderUpDown = 0
+
+        task = self.defaultTask
+        task.config = config
+
+        defects = task._findHotAndColdPixels(self.flatExp)
+
+        allBBoxes = self.darkBBoxes + self.brightBBoxes
+
+        boxesMeasured = []
+        for defect in defects:
+            boxesMeasured.append(defect.getBBox())
+
+        for expectedBBox in allBBoxes:
+            self.assertIn(expectedBBox, boxesMeasured)
+
+    def test_defectFindingEdgeIgnore(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderUpDown = 0
+        task = self.defaultTask
+        task.config = config
+        defects = task._findHotAndColdPixels(self.flatExp)
+
+        shouldBeFound = self.darkBBoxes[self.noEdges] + self.brightBBoxes[self.noEdges]
+
+        boxesMeasured = []
+        for defect in defects:
+            boxesMeasured.append(defect.getBBox())
+
+        for expectedBBox in shouldBeFound:
+            self.assertIn(expectedBBox, boxesMeasured)
+
+        shouldBeMissed = self.darkBBoxes[self.onlyEdges] + self.brightBBoxes[self.onlyEdges]
+        for boxMissed in shouldBeMissed:
+            self.assertNotIn(boxMissed, boxesMeasured)
```

Comment:
How are the tests above here any different from the tests in the original file?  Isn't it only the value tests below that matter?  Or am I missing something?

Reply:
They are mostly the same, as I mentioned in another comment above: `I added a new file for the combined defects task that repeats the tests in the existing defects task and added a couple of tests to the VALUE option.`

Reply:
But it looks as if they are identical, so what's different?  I'm trying to figure out if anything is different except for the couple of VALUE tests.  If so, then those are the only tests that are needed here and the duplicated tests should be deleted.  But if something else is different, please let me know!

Reply:
They are mostly identical. The default class and configurations are the combined ones, e.g., `self.defaultTask = cpPipe.defects.MeasureDefectsCombinedTask()`, in addition to the `VALUE` tests.

Reply:
The task is different, but it's a sub-task, and the tests here are only calling the individual methods, and not the task.  So the code paths are essentially identical.  I would remove all of these except for the VALUE tests which are new and useful.  And the combined part of the pipeline (changing connections) is done with ci_cpp_gen3.

Reply:
Sounds good. I have implemented and pushed these changes.